stacks-common/build.rs

```diff
@@ -34,6 +90,40 @@ fn main() {
         }
     }
 
+    if let Some(git) = current_git_hash() {
+        // println!("git commit: {}", git);
+        rust_code.push_str(&format!(
+            "pub const GIT_COMMIT: Option<&'static str> = Some(\"{}\");\n",
+            git
+        ));
+        println!("cargo:rustc-env=GIT_COMMIT={}", git);
+    } else {
+        rust_code.push_str(&format!(
+            "pub const GIT_COMMIT: Option<&'static str> = None;\n"
+        ));
+    }
+    if let Some(git) = current_git_branch() {
+        rust_code.push_str(&format!(
+            "pub const GIT_BRANCH: Option<&'static str> = Some(\"{}\");\n",
+            git
+        ));
+        println!("cargo:rustc-env=GIT_BRANCH={}", git);
+    } else {
+        rust_code.push_str(&format!(
+            "pub const GIT_BRANCH: Option<&'static str> = None;\n"
+        ));
+    }
```

Comment:
I think you could just use the debug print to create the `Option` value, so you don't have to duplicate these variable declarations. For example:

```suggestion
    rust_code.push_str(
        &format!("pub const GIT_BRANCH: Option<&'static str> = {git:?});\n")       
    );
    if let Some(git) = current_git_branch() {
        println!("cargo:rustc-env=GIT_BRANCH={git}");
    }
```

Also, it's a nit, but IMO it's easier to read if you inline the variables into the format strings

Reply:
Brilliant idea of using the debug value inlined instead of `Some("{}")`!